### PR TITLE
Fix a CORS test error

### LINKED ISSUE #26849: /cors/basic.htm "Same domain basic usage" fails in the same way on all browsers
Behold: https://wpt.fyi/results/cors/basic.htm?run_id=5681964860309504&run_id=5636499091488768&run_id=5635783039909888&run_id=5195490306031616

In all browsers, it appears that this assert failed:

https://github.com/web-platform-tests/wpt/blob/6ea3164d34a632b16b857d7474985b93acd8de11/cors/basic.htm#L44

@annevk is it actually the test that is wrong here?

### PR DESCRIPTION
**Pull request by yutakahirano**
The following statement throws when "origin" is specified neither
via URL query nor request headers.

```
request.GET.first(b"origin", request.headers.get(b'origin'))
```

Fix that by using "none" as the default value.

Clean up cors/basic.htm.

Fixes #26849.

### CODE REVIEW

File: cors/basic.htm [hunk 1]
```diff
@@ -6,48 +6,43 @@
...
 function cors(desc, scheme, subdomain, port) {
-    if (!scheme) {
-        var url = "";
-    } else {
-        if (!port) {
-            port = location.port;
-        }
-        var url = scheme + "://" + (subdomain ? subdomain + "." : "") + location.hostname + ":" + port + dirname(location.pathname)
+    port = port || location.port;
+    subdomain = subdomain || "";
```
Comment (annevk) [inline cors/basic.htm:16]: Nit: you can use `port = location.port` and `subdomain = ""` in the function declaration directly these days I think, that might be nicer.

File: cors/basic.htm [hunk 2]
```diff
@@ -6,48 +6,43 @@
...
-            client.onreadystatechange = this.step_func(function(e) {
-                // Second request, test that it passes with the allowed-origin
-                if (client.readyState < 4) return;
-                assert_true(client.response.indexOf("hest_er_best") != -1, "Got CORS-allowed response");
-                this.done();
-            });
-            client.send();
+    async_test((t) => {
+        const client = new XMLHttpRequest();
+        const url = `${base}resources/cors-makeheader.py?get_value=hest_er_best`
```
Comment (annevk) [inline cors/basic.htm]: It seems this variable is no longer used?

File: cors/basic.htm [hunk 3]
```diff
@@ -6,48 +6,43 @@
...
+            assert_true(client.response.includes("hest_er_best"), "Got response");
+        });
+        client.onerror = t.step_func_done(() => {
+            assert_false(sameorigin, "Same origin request must be accepted.");
+        });
+    }, `${desc}, origin: none`);
+
+    async_test((t) => {
+        const client = new XMLHttpRequest();
+        const url = `${base}resources/cors-makeheader.py?get_value=hest_er_best`
```
Comment (annevk) [inline cors/basic.htm]: Same.

File: cors/resources/cors-makeheader.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 from wptserve.utils import isomorphic_decode
 
 def main(request, response):
-    origin = request.GET.first(b"origin", request.headers.get(b'origin'))
+    origin = request.GET.first(b"origin", request.headers.get(b'origin') or b'none')
```
Comment (annevk) [inline cors/resources/cors-makeheader.py:6]: When would we need the fallback here? Why doesn't the second request include the origin header?

Comment (yutakahirano) [reply cors/resources/cors-makeheader.py:6]: Because same-origin requests doesn't have an origin header.

File: cors/basic.htm (hunk 2 above)
Comment (yutakahirano) [reply cors/basic.htm]: Thanks, removed.

File: cors/basic.htm (hunk 3 above)
Comment (yutakahirano) [reply cors/basic.htm]: Removed.

File: cors/basic.htm (hunk 1 above)
Comment (yutakahirano) [reply cors/basic.htm]: Done.

File: cors/resources/cors-makeheader.py (hunk 4 above)
Comment (yutakahirano) [reply cors/resources/cors-makeheader.py:6]: Rephrasing:

> When would we need the fallback here?

The "origin: echo" test where _sameorigin_ is true.

> Why doesn't the second request include the origin header?

Do you mean the origin component in the request URL query?
I think the test intention is:

- The request fails when access-control-allow-origin header is missing except for the same origin case (the first test case).
- The request succeeds when access-control-allow-origin header is correctly set (the second test case).

So, I think attaching "&origin=${document.origin}" to the URL of the second request would fix the failure without changing the python script. Do you prefer the solution?

Comment (annevk) [reply cors/resources/cors-makeheader.py:6]: I guess slightly, but I'm okay with this too given it's an old test.

**Review by annevk — APPROVED** [review]

**Comment by foolip** [comment]
Thanks for fixing, @yutakahirano!

@annevk since you approved I guess it's OK to merge this, so I'll do just that.

**Pull request merged**